New module: azure_rm_eventhub

saurabh3796 opened the pull request:
##### SUMMARY
This module introduces a wide variety of configurations for creating a Event Hub Namespace and Event Hub on Azure.
Create Event Hub Namespace on Azure.
Create Event Hub on Azure.
Update Event Hub Namespace on Azure.
Update Event Hub on Azure.
Delete Event Hub Namespace on Azure.
Delete Event Hub on Azure.
Get Event Hub Namespace on Azure.
Get Event Hub on Azure.
##### ISSUE TYPE
New Module Pull Request

##### COMPONENT NAME
azure_rm_eventhub
##### ADDITIONAL INFORMATION
This module allows users to create/update or delete the Event Hub Namespace and Event Hub on Azure.
It also provides all the parameters given by:
https://docs.microsoft.com/en-us/azure/templates/microsoft.eventhub/namespaces?tabs=json
https://docs.microsoft.com/en-us/azure/templates/microsoft.eventhub/namespaces/eventhubs?tabs=json

##### Example,
```  
azure_rm_eventhub:
    resource_group: testgroupans
    location: eastus
    namespace_name: myNamespace
    tags:
        a: b
    sku: Basic

```

smalpani13 committed:
Eventhub CRUD operations

saurabh3796 committed:
Merge branch 'dev' into event_hub

saurabh3796 commented:
@Fred-sun Can you please review this PR for new module, Thanks.

Fred-sun commented:
> @Fred-sun Can you please review this PR for new module, Thanks.

Thank you for your contribution, under review!

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+        required: if state is present
```
The location parameter is not required, We can set it same with resource group location when not set location.  as following method!
```e.g.
        resource_group = self.get_resource_group(self.resource_group)
        if not self.location:
            # Set default location
            self.location = resource_group.location

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+partition_count_list = [i for i in range(1, 33)]
+
+
+class AzureEventHub(AzureRMModuleBase):
```
```suggestion
class AzureRMEventHub(AzureRMModuleBase):
```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,240 @@
+    pass
+
+
+class AzureEventHubInfo(AzureRMModuleBase):
```
```suggestion
class AzureRMEventHubInfo(AzureRMModuleBase):
```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+
+        self.module_arg_spec = dict(
+            message_retention_in_days=dict(type='int'),
+            name=dict(type='str'),
```
Argument 'name' is required in document. the argument spec is not required

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+            name=dict(type='str', required=True)
+        )
+
+        self.module_arg_spec = dict(
```
The argument spec has defined ['message_retention_in_days', 'status', 'partition_count'], but missing in document!

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+          - absent
+          - present
+      type: str
+    
```
contains whitespace

Fred-sun commented on plugins/module_utils/azure_rm_common.py:
```diff
@@ -1325,6 +1333,15 @@ def datalake_store_client(self):
+                base_url=self._cloud_environment.endpoints.resource_manager,
+                api_version='2018-05-04')
+        return self._event_hub_client
 
```
expect 2 bank line!

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+        "created_at": "2021-04-29T10:05:20.377Z",
+        "type": "Microsoft.eventHubs/namespaces"
+    }
+    
```
contains whitespace

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+        self.state = None
+
+        super(AzureEventHub, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                   supports_check_mode=True,
```
lines 190, 191 continuation line over-indented for visual indent!

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+    # This is handled in azure_rm_common
+    pass
+
+partition_count_list = [i for i in range(1, 33)]
```
unnecessary-comprehension: Unnecessary use of a comprehension

saurabh3796 replied:
Allowed values for partition count is ranging from 1 to 32. If we mention it arg spec like [1, 2, 3 .. 32] it will not look good.
Can you point me out to any alternative way to achieve this one ?

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
```
keep

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,405 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
```
Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,240 @@
+            - The name of the Event hub.
+        type: str
+extends_documentation_fragment:
+- azure.azcollection.azure
```
```suggestion
    - azure.azcollection.azure
```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,240 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
```
Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN.

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,240 @@
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
```
keep

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,240 @@
+        type: str
+    namespace_name:
+            - The name of the namspace
+        required: True
```
```suggestion
        required: True
```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,240 @@
+    returned: always
+    type: dict
+    sample:{
+            "additional_properties": {},
```
```suggestion
            "additional_properties": {},
```

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+`- name: Create random event hub and namespace
```
```suggestion
- name: Create random event hub and namespace
```

Fred-sun commented:
@saurabh3796 Please add following line to ignore2.11.txt, 2.9.txt, 2.10.txt and 2.12.txt. Thank you very much!
```e.g.
plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-unknown-key

saurabh3796 committed:
Update azure_rm_common.py

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 committed:
Update azure_rm_eventhub_info.py

saurabh3796 committed:
Update ignore-2.10.txt

saurabh3796 committed:
Update ignore-2.11.txt

saurabh3796 committed:
Update ignore-2.12.txt

saurabh3796 committed:
Update ignore-2.9.txt

saurabh3796 committed:
Update main.yml

saurabh3796 committed:
Merge branch 'dev' into event_hub

Fred-sun commented:
@saurabh3796 Please help resolve these confilict files! Thank you very much!

saurabh3796 committed:
Merge branch 'dev' into event_hub

saurabh3796 commented:
@Fred-sun Resolved the conflicts, thanks for review.

saurabh3796 committed:
Merge branch 'dev' into event_hub

saurabh3796 commented:
@Fred-sun Can you please review this PR whenever you are available. Thanks.

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,421 @@
+
+DOCUMENTATION = '''
+---
+module: azure_rm_eventhub
```
```suggestion
module: azure_rm_eventhub
version_added: "1.6.0"
```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,239 @@
+
+DOCUMENTATION = '''
+---
+module: azure_eventhub_info
```
```suggestion
module: azure_eventhub_info
version_added: "1.6.0"
```

Fred-sun reviewed (commented):
@saurabh3796 Please help to delete these ignores on ignore2.11.txt, 2.9.txt, 2.10.txt and 2.12.txt. It has been fixed by #540!

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,239 @@
+module: azure_eventhub_info
+short_description: Get Azure Event Hub
+description:
+    - Get facts of Azure Event Hub
```
```suggestion
    - Get facts of Azure Event Hub.
```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,239 @@
+        required: True
+        type: str
+    namespace_name:
+            - The name of the namspace
```
```suggestion
            - The name of the namspace.
```

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 committed:
Update azure_rm_eventhub_info.py

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,421 @@
+        required: False
+        type: str
+    message_retention_in_days:
+        description: 
```
```suggestion
        description:
```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,421 @@
+        description:
+            - Number of partitions created for the Event Hub.
+        required: False
+        type: int
```
missing choices!

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,421 @@
+        description:
+            - Enumerates the possible values for the status of the Event hub.
+        required: False
+        type: str
```
missing choices and default value!

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,239 @@
+            - The name of the resource group.
+        required: True
+        type: str
+    namespace_name:
```
```suggestion
    namespace_name:
        description:
```

saurabh3796 committed:
Update azure_rm_eventhub_info.py

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+    namespace_name: "{{ namespace_name }}"
+    resource_group: "{{ resource_group }}"
+    sku: "Basic"
+    check_mode: yes
```
```suggestion
  check_mode: yes
```

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+    name: "{{ name }}"
+    resource_group: "{{ resource_group }}"
+    sku: "Basic"
+    check_mode: yes
```
```suggestion
  check_mode: yes
```

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+    that: results.changed
+
+
+- name: Create Event Hub Namespace (check mode)
```
```suggestion
- name: Create Event Hub Namespace
```

saurabh3796 committed:
Update azure_rm_eventhub.py

Fred-sun commented:
@saurabh3796 If you resolve the comment! Please mark “resolve conversation” Thank you very much!

saurabh3796 committed:
Update main.yml

saurabh3796 committed:
Update main.yml

saurabh3796 commented:
@Fred-sun Addressed the changes suggested. Thanks.

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,434 @@
+            - Number of partitions created for the Event Hub.
+        required: False
+        type: int
+        choices: Choose value between 1 to 32(both included).
```
According to Ansible Sanity definition specification should be a list of 1~32! Thank you very much!
```suggestion
        choices:  [1,2,3,4,5,6,7,8,......,31,32]
```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,434 @@
+    # This is handled in azure_rm_common
+    pass
+
+partition_count_list = [i for i in range(1, 33)]
```
According to Ansible Sanity definition specification should be a list of 1~32! Thank you very much!
```suggestion
partition_count_list = [1,2,3,4,5,6,7,8,......,31,32]
```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,241 @@
+    namespace_name:
+        description:
+            - The name of the namspace.
+        required: True
```
Same with argument_spec
```suggestion
        required: False
```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,241 @@
+
+DOCUMENTATION = '''
+---
+module: azure_eventhub_info
```
```suggestion
module: azure_rm_eventhub_info
```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,241 @@
+        - Current state of the Event Hub namesapce or Event Hub.
+    returned: always
+    type: dict
+    sample:{
```
```suggestion
    sample: {
```

Fred-sun commented on tests/sanity/ignore-2.9.txt:
```diff
@@ -181,5 +181,9 @@ plugins/modules/azure_rm_adpassword_info.py validate-modules:nonexistent-paramet
 plugins/modules/azure_rm_adpassword_info.py validate-modules:import-before-documentation
 plugins/modules/azure_rm_adserviceprincipal.py validate-modules:nonexistent-parameter-documented
 plugins/modules/azure_rm_adserviceprincipal_info.py validate-modules:nonexistent-parameter-documented
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.9.txt:
```diff
@@ -181,5 +181,9 @@ plugins/modules/azure_rm_adpassword_info.py validate-modules:nonexistent-paramet
 plugins/modules/azure_rm_adserviceprincipal.py validate-modules:nonexistent-parameter-documented
 plugins/modules/azure_rm_adserviceprincipal_info.py validate-modules:nonexistent-parameter-documented
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.9.txt:
```diff
@@ -181,5 +181,9 @@ plugins/modules/azure_rm_adpassword_info.py validate-modules:nonexistent-paramet
 plugins/modules/azure_rm_adserviceprincipal_info.py validate-modules:nonexistent-parameter-documented
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.9.txt:
```diff
@@ -181,5 +181,9 @@ plugins/modules/azure_rm_adpassword_info.py validate-modules:nonexistent-paramet
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.12.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.12.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.12.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.12.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.12.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.12.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.12.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_vmbackuppolicy.py validate-modules:parameter-list-no-elements
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.12.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_subnet.py validate-modules:doc-elements-mismatch
 plugins/modules/azure_rm_vmbackuppolicy.py validate-modules:parameter-list-no-elements
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.11.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.11.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.11.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.11.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.11.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.11.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.11.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_vmbackuppolicy.py validate-modules:parameter-list-no-elements
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.11.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_subnet.py validate-modules:doc-elements-mismatch
 plugins/modules/azure_rm_vmbackuppolicy.py validate-modules:parameter-list-no-elements
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.10.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.10.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub_info.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.10.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.10.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_enventhub.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.10.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.10.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_privatednsrecordset_info.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.10.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_vmbackuppolicy.py validate-modules:parameter-list-no-elements
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-unknown-key
```
```suggestion

```

Fred-sun commented on tests/sanity/ignore-2.10.txt:
```diff
@@ -354,5 +354,13 @@ plugins/modules/azure_rm_openshiftmanagedcluster.py validate-modules:parameter-l
 plugins/modules/azure_rm_subnet.py validate-modules:doc-elements-mismatch
 plugins/modules/azure_rm_vmbackuppolicy.py validate-modules:parameter-list-no-elements
 plugins/modules/azure_rm_privatednsrecordset.py validate-modules:invalid-ansiblemodule-schema
+plugins/modules/azure_rm_privatednsrecordset.py validate-modules:required_if-requirements-unknown
```
```suggestion

```

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 committed:
Update azure_rm_eventhub_info.py

saurabh3796 committed:
Update ignore-2.9.txt

saurabh3796 committed:
Update ignore-2.12.txt

saurabh3796 committed:
Update ignore-2.11.txt

saurabh3796 committed:
Update ignore-2.10.txt

saurabh3796 commented:
@Fred-sun Addressed the changes and mark resolved conversation. Thanks.

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,434 @@
+        type: int
+    partition_count:
+        description:
+            - Number of partitions created for the Event Hub.
```
```suggestion
            - Number of partitions created for the Event Hub.
            - Range from 1 to 32.
```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,434 @@
+            - Number of partitions created for the Event Hub.
+        required: False
+        type: int
+        choices: [1,2,3,4,5,6,7,8,......,31,32]
```
```suggestion

```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,434 @@
+    # This is handled in azure_rm_common
+    pass
+
+partition_count_list = [1,2,3,4,5,6,7,8,......,31,32]
```
```suggestion

```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,434 @@
+            message_retention_in_days=dict(type='int'),
+            name=dict(type='str'),
+            namespace_name=dict(type='str', required=True),
+            partition_count=dict(type='int', choices=partition_count_list),
```
```suggestion
            partition_count=dict(type='int'),
```

saurabh3796 committed:
Update azure_rm_eventhub.py

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,432 @@
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
```
Please move line 161~164 to the outside of the try statement! Thank you very much!

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,241 @@
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
```
Please move line 76 to the outside of the try statement! Thank you very much!

Fred-sun replied:
```suggestion

```

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,241 @@
+    }
+'''
+
+try:
```
```suggestion
from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
try:
```

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 committed:
Update azure_rm_eventhub_info.py

Fred-sun commented on plugins/modules/azure_rm_eventhub_info.py:
```diff
@@ -0,0 +1,237 @@
+    }
+'''
+
+from msrestazure.azure_exceptions import CloudError
```
```suggestion
try:
    from msrestazure.azure_exceptions import CloudError
except ImportError:
    # This is handled in azure_rm_common
    pass
```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,429 @@
+'''
+
+
+from msrestazure.azure_exceptions import CloudError
```
```suggestion
try:
    from msrestazure.azure_exceptions import CloudError
except ImportError:
    # This is handled in azure_rm_common
    pass
```

saurabh3796 committed:
Update azure_rm_eventhub_info.py

Fred-sun commented:
@saurabh3796 We only need to do “try --- except"  for msrestazure related modules, no other need, thank you!

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 committed:
Update azure_rm_eventhub_info.py

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,432 @@
+    # This is handled in azure_rm_common
+    pass
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+from azure.mgmt.eventhub.models import Eventhub, EHNamespace
```
```suggestion

```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,432 @@
+    pass
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+from azure.mgmt.eventhub.models import Eventhub, EHNamespace
+from azure.mgmt.eventhub.models.sku import Sku
```
```suggestion

```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,432 @@
+'''
+
+try:
+    from msrestazure.azure_exceptions import CloudError
```
```suggestion
    from msrestazure.azure_exceptions import CloudError
    from azure.mgmt.eventhub.models import Eventhub, EHNamespace
    from azure.mgmt.eventhub.models.sku import Sku
```

Fred-sun reviewed (commented):
@saurabh3796 Sorry, missing azure module! We only need to do “try --- except" for msrestazure and azure related modules,  thank you!

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 commented:
@Fred-sun  Updated the suggested changes and tested it, forgot one condition added now. Thanks for thorough checking otherwise missed that one.

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+      namespace_name: "test{{ resource_group | hash('md5') | truncate(16, True, '') + (65535 | random | string) }}"
+      name: "test{{ resource_group | hash('md5') | truncate(16, True, '') + (65535 | random | string) }}"
+
+
```
```suggestion
```

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+- assert:
+    that: results.changed
+
+
```
```suggestion
```

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+
+- assert:
+    that: results.changed
+
```
```suggestion
```

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+      - results.state.tags.test == 'modified'
+
+- name: Retrieve Namespace
+  azure_rm_eventhub:
```
```suggestion
  azure_rm_eventhub_info:
```

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+
+- name: Retrieve Namespace
+  azure_rm_eventhub:
+    location: eastus2
```
```suggestion
```

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,152 @@
+- name: Assert that facts module returned result
+  assert:
+    that: 
+      - results.ddos_protection_plan[0].tags.test == 'modified'
```
```suggestion
      - results.namespace[0].tags.test == 'modified'
```

Fred-sun reviewed (commented):
@saurabh3796 Please help execute the task（tests/integration/targets/azure_rm_eventhub/tasks/main.yml） to make sure the test case can be used, thank you!

saurabh3796 committed:
Update main.yml

saurabh3796 commented:
I have run them manually in playbook.
`- name: Delete Namespace
  azure_rm_eventhub:
    location: eastus2
    namespace_name: "nsCreateTest"
    resource_group: "newctvm"
    state: absent

changed: [localhost] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": "auto",
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "message_retention_in_days": null,
            "name": null,
            "namespace_name": "nsCreateTest",
            "partition_count": null,
            "password": null,
            "profile": null,
            "resource_group": "newctvm",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "sku": "Basic",
            "state": "absent",
            "status": "Active",
            "subscription_id": "xxxxx",
            "tags": null,
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": {},
        "created_at": "2021-05-27T05:01:06.873Z",
        "critical": null,
        "data_center": null,
        "enabled": null,
        "is_auto_inflate_enabled": false,
        "location": "eastus2",
        "maximum_throughput_units": 0,
        "metric_id": "xxxxx",
        "name": "nsCreateTest",
        "namespace_type": null,
        "provisioning_state": "Succeeded",
        "region": null,
        "scale_unit": null,
        "service_bus_endpoint": "xxxx",
        "sku": "Basic",
        "status": "Deleted",
        "tags": {
            "test": "modified"
        },
        "type": "Microsoft.EventHub/Namespaces",
        "updated_at": "2021-05-27T05:06:35.170Z"
    }
}
`

saurabh3796 commented:
`- name: Update Namespace
  azure_rm_eventhub:
    location: eastus2
    namespace_name: "nsCreateTest"
    resource_group: "newctvm"
    tags:
      test: modified
  register: results

- assert:
    that: 
      - results.changed
      - results.state.tags.test == 'modified'

changed: [localhost] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": "auto",
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "message_retention_in_days": null,
            "name": null,
            "namespace_name": "nsCreateTest",
            "partition_count": null,
            "password": null,
            "profile": null,
            "resource_group": "newctvm",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "sku": "Basic",
            "state": "present",
            "status": "Active",
            "subscription_id": "xxxx",
            "tags": {
                "test": "modified"
            },
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": {},
        "created_at": "2021-05-27T05:01:06.873Z",
        "critical": null,
        "data_center": null,
        "enabled": null,
        "is_auto_inflate_enabled": false,
        "location": "eastus2",
        "maximum_throughput_units": 0,
        "metric_id": "xxx",
        "name": "nsCreateTest",
        "namespace_type": null,
        "provisioning_state": "Succeeded",
        "region": null,
        "scale_unit": null,
        "service_bus_endpoint": "xxx",
        "sku": "Basic",
        "tags": {
            "test": "modified"
        },
        "type": "Microsoft.EventHub/Namespaces",
        "updated_at": "2021-05-27T05:06:35.170Z"
    }
}
`

saurabh3796 commented:
@Fred-sun  All tests are running fine, attached results for couple of ones.

Fred-sun commented:
> @Fred-sun All tests are running fine, attached results for couple of ones. Thanks

@saurabh3796 Update event_hub fail! thanks!

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 commented:
Update event hub
playbook:
`---
    - hosts: localhost
      connection: local
      tasks:
       - name: Retrieve event hub
         azure.azcollection.azure_rm_eventhub:
           resource_group: newctvm
           state: present
           location: eastus2
           status: "Active"
           message_retention_in_days: 2
           sku: "Standard"
           namespace_name: nsCreateTest
           name: eventCreateTest
         register: results`
         
Result - 
`ok: [localhost] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": "auto",
            "cert_validation_mode": null,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cloud_environment": "AzureCloud",
            "location": "eastus2",
            "message_retention_in_days": 2,
            "name": "eventCreateTest",
            "namespace_name": "nsCreateTest",
            "partition_count": null,
            "password": null,
            "profile": null,
            "resource_group": "newctvm",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "sku": "Standard",
            "state": "present",
            "status": "Active",
            "subscription_id": "xxxxxxx",
            "tags": null,
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
        }
    },
    "state": {
        "additional_properties": {
            "location": "East US 2"
        },
        "created_at": "2021-05-27T09:08:35.813Z",
        "message_retention_in_days": 2,
        "name": "eventCreateTest",
        "partition_count": 4,
        "partition_ids": [
            "0",
            "1",
            "2",
            "3"
        ],
        "status": "Active",
        "tags": null,
        "updated_at": "2021-05-27T10:07:00.867Z"
    }
}
`

saurabh3796 committed:
Update main.yml

saurabh3796 commented:
> Why do you put it in a try statement?

If eventhub or namespace does not exist, it raise an exception in sdk.

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 committed:
Update azure_rm_eventhub.py

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,437 @@
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
```
Move "from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase" to outside try statement!

saurabh3796 replied:
Yes updated it.

Fred-sun commented:
@saurabh3796 Failed when testing for idempotency! Thank you very much!

saurabh3796 commented:
@Fred-sun Can you share the steps to run integration tests, I am trying to run same on local but not able to run.

Fred-sun commented:
> @Fred-sun Can you share the steps to run integration tests, I am trying to run same on local but not able to run.

Execute the tasks,

```e.g.
---
- name: Using Azure collection
  hosts: localhost
  connection: local
  collections:
    - azure.azcollection
  vars:
    resource_group: v-xisuRG
  tasks:
    - name: create resource gorup
      azure_rm_resourcegroup:
        name: v-xisuRG
        location: eastus
    - include: main.yml
~

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,138 @@
+
+- assert:
+    that: 
+      - not results.changed
```
Since we are checking the update results, we should return results.changed is True
```suggestion
      - results.changed
```

saurabh3796 committed:
Update azure_rm_eventhub.py

saurabh3796 commented:
@Fred-sun  Ran all integration tests successfully, attached the result. Thanks.

saurabh3796 committed:
Update main.yml

Fred-sun commented on tests/integration/targets/azure_rm_eventhub/tasks/main.yml:
```diff
@@ -0,0 +1,142 @@
+
+- assert:
+    that:
+      - results.changed
```
Since this is idempotent, it should return results.changed is False

```suggestion
      - not results.changed
```

saurabh3796 commented:
New updated integration tests results - 
`TASK [Gathering Facts] *************************************************************************************************************************************************************************************
ok: [localhost]

TASK [create resource gorup] *******************************************************************************************************************************************************************************
ok: [localhost]

TASK [Create random event hub and namespace] ***************************************************************************************************************************************************************
ok: [localhost]

TASK [Create Event Hub Namespace (check mode)] *************************************************************************************************************************************************************
changed: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create Event Hub  (check mode)] **********************************************************************************************************************************************************************
changed: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create Event Hub Namespace] **************************************************************************************************************************************************************************
changed: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Create Event Hub] ************************************************************************************************************************************************************************************
changed: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Update Namespace] ************************************************************************************************************************************************************************************
changed: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Update Event Hub] ************************************************************************************************************************************************************************************
changed: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Retrieve Namespace] **********************************************************************************************************************************************************************************
ok: [localhost]

TASK [Test idempotent] *************************************************************************************************************************************************************************************
ok: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Delete Event Hub] ************************************************************************************************************************************************************************************
changed: [localhost]

TASK [Delete Event Hub (idempotent)] ***********************************************************************************************************************************************************************
ok: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Delete Namespace] ************************************************************************************************************************************************************************************
changed: [localhost]

TASK [Pause for 1 minutes to check namespace deleted] ******************************************************************************************************************************************************
Pausing for 60 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
ok: [localhost]

TASK [Delete Namespace (idempotent)] ***********************************************************************************************************************************************************************
ok: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=25   changed=8    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0`

saurabh3796 committed:
Update main.yml

saurabh3796 committed:
Update azure_rm_eventhub.py

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,439 @@
+            # don't change anything if creating an existing namespace, but change if deleting it
+            if self.state == 'present':
+                changed = False
+                
```
```suggestion

```

Fred-sun commented on plugins/modules/azure_rm_eventhub.py:
```diff
@@ -0,0 +1,439 @@
+                if update_tags:
+                    changed = True
+                elif self.namespace_name and not self.name:
+                    
```
```suggestion

```

saurabh3796 committed:
Update azure_rm_eventhub.py

haiyuazhang committed:
Merge branch 'dev' into event_hub
